src/components/datagridField/util.js

```diff
@@ -34,15 +34,17 @@ function chunkConditional(array, predicate) {
 }
 
 function buildVariableSizeFieldSection(columns) {
-  const filteredColumns = columns.filter((column) => (!column.meta || !column.meta.hidden));
```

Comment:
Keep the filter logic to make sure columns without meta are excluded otherwise code will break when config is malformed.